set.go

```diff
@@ -438,3 +438,38 @@ func (s *Set) registerMetric(name string, m metric) {
 		panic(fmt.Errorf("BUG: metric %q is already registered", name))
 	}
 }
+
+// UnregisterMetric will remove a registered metric
+func (s *Set) UnregisterMetric(name string) {
+	if err := validateMetric(name); err != nil {
+		panic(fmt.Errorf("BUG: invalid metric name %q: %s", name, err))
+	}
+	s.mu.Lock()
+	nm, ok := s.m[name]
+	if ok {
+		delete(s.m, name)
+		for i, a := range s.a {
+			if a == nm {
+				s.a = append(s.a[:i], s.a[i+1:]...)
+			}
+		}
+	}
+	s.mu.Unlock()
+}
+
+// HasMetric will return true, if the metric exists
+func (s *Set) HasMetric(name string) bool {
+	s.mu.Lock()
+	_, ok := s.m[name]
+	s.mu.Unlock()
+	return ok
+}
+
+// ListMetricNames will return a list of all registered metrics
+func (s *Set) ListMetricNames() []string {
```

Comment:
What is the use case for this method?

Reply:
Our use case is that we need to find all metrics belonging to a specific process being identified by a common prefix in the metric names that need to be unregistered when this process cannot be queried anymore through it's API. The existing `Set` package practically does everything needed here. 

So this completes the functionality of the Set to be able to add, remove and list items of the set.

A more generic `List()` method could also be used like this:

```go
// Remove all metrics starting with "dummy_"
for name, _ := range myset.List() {
    if (strings.HasPrefix(name, "dummy_")) {
        myset.UnregisterMetric(name)
    }
}
```

So any form of being able to list the names of the metrics are required for our use case.

Reply:
OK, let's keep this method here.